### PR TITLE
RTCIceCandidatePairStats - updates

### PR DESCRIPTION
**Pull request by hamishwillee**
Updates [`RTCIceCandidatePairStats`](https://developer.mozilla.org/en-US/docs/Web/API/RTCIceCandidatePairStats) to current MDN templates.

This is in three commits. 
- The first adds layout only changes (and can be ignored)
- The second tidies existing properties to current templates.
- The third adds docs for `bytesDiscardedOnSend`, `consentRequestsSent`, `packetsReceived`, `packetsSent` properties which are in BCD but not in the docs.

Note that the new docs are not much more than could be inferred from the specification. 

Note, 
- `availableIncomingBitrate` is documented but not in BCD. The collector is not reliable in respect to this API so they might be present just not recorded.
- There are some non-standard methods which aren't in BCD either : readable, writable, selected 
- I have ignored ^^^ - this is better than it was.

This is part of https://github.com/mdn/mdn/issues/384

### CODE REVIEW

File: files/en-us/web/api/rtcicecandidatepairstats/bytesdiscardedonsend/index.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+title: "RTCIceCandidatePairStats: bytesDiscardedOnSend property"
+short-title: bytesDiscardedOnSend
+slug: Web/API/RTCIceCandidatePairStats/bytesDiscardedOnSend
+page-type: web-api-instance-property
+browser-compat: api.RTCStatsReport.type_candidate-pair.bytesDiscardedOnSend
+---
+
+{{APIRef("WebRTC")}}
+
+The **`bytesDiscardedOnSend`** property of the {{domxref("RTCIceCandidatePairStats")}} dictionary indicates the total number of bytes discarded due to socket errors
```
Comment (wbamberg) [inline files/en-us/web/api/rtcicecandidatepairstats/bytesdiscardedonsend/index.md:11]:
```suggestion
The **`bytesDiscardedOnSend`** property of the {{domxref("RTCIceCandidatePairStats")}} dictionary indicates the total number of bytes discarded due to socket errors.
```

**Review by wbamberg — COMMENTED** [review]
"A number value" reads weirdly to me compared with just "A number". But then we also say things like "An integer value" and "A Boolean value", IDK, just nit-picking really. Please accept or not as you choose.

We also say "a real number" in a couple of places, is this significantly different to "a number"?

File: files/en-us/web/api/rtcicecandidatepairstats/availableincomingbitrate/index.md [hunk 2]
```diff
@@ -8,37 +8,21 @@ browser-compat: api.RTCIceCandidatePairStats.availableIncomingBitrate
...
-{{domxref("RTCIceCandidatePairStats.availableOutgoingBitrate", "availableOutgoingBitrate")}}.
+You can get the outgoing available bitrate from {{domxref("RTCIceCandidatePairStats.availableOutgoingBitrate", "availableOutgoingBitrate")}}.
 
-## Syntax
+## Value
 
-```js-nolint
-availableIncomingBitrate = rtcIceCandidatePairStats.availableIncomingBitrate
-```
+A number value which approximates the amount of available bandwidth for incoming data on the network connection described by the `RTCIceCandidatePair`.
```
[the remark this answers is not in the document]
Comment (hamishwillee) [reply files/en-us/web/api/rtcicecandidatepairstats/availableincomingbitrate/index.md:18]: Works for me.

File: files/en-us/web/api/rtcicecandidatepairstats/currentroundtriptime/index.md [hunk 3]
```diff
@@ -8,28 +8,14 @@ browser-compat: api.RTCStatsReport.type_candidate-pair.currentRoundTripTime
...
-candidates.
+The **`currentRoundTripTime`** property of the {{domxref("RTCIceCandidatePairStats")}} dictionary indicates the number of seconds it takes for data to be sent by this peer to the remote peer and back over the connection described by this pair of {{Glossary("ICE")}} candidates.
 
-## Syntax
+## Value
 
-```js-nolint
-rtt = rtcIceCandidatePairStats.currentRoundTripTime
-```
+A real number indicating the round-trip time, in seconds, for the connection described by the pair of candidates for which this `RTCIceCandidatePairStats` object offers statistics.
```
Comment (hamishwillee) [inline files/en-us/web/api/rtcicecandidatepairstats/currentroundtriptime/index.md:15]:
```suggestion
A number indicating the round-trip time, in seconds, for the connection described by the pair of candidates for which this `RTCIceCandidatePairStats` object offers statistics.
```

**Comment by hamishwillee** [comment]
> "A number value" reads weirdly to me compared with just "A number". But then we also say things like "An integer value" and "A Boolean value", IDK, just nit-picking really. Please accept or not as you choose.

Agree. Accepted all.

> We also say "a real number" in a couple of places, is this significantly different to "a number"?

I'll purge "real" because we don't say so elsewhere - a number is generally accepted to mean "real number"/"not an integer".

----

TLDR: 

So why did I do this. Consider (https://github.com/mdn/content/pull/36527#discussion_r1825300015)

> A real number indicating the round-trip time, in seconds, for the connection described by the pair of candidates for which this `RTCIceCandidatePairStats` object offers statistics.

This is now 

> A number indicating the round-trip time, in seconds, for the connection described by the pair of candidates for which this `RTCIceCandidatePairStats` object offers statistics.

Tell me, would you assume that the value could include fractional seconds? 
The rule says we should, and use integer if we don't. But I felt a little uncomfortable.

File: files/en-us/web/api/rtcicecandidatepairstats/index.md [hunk 4]
```diff
@@ -13,38 +13,54 @@ The statistics can be obtained by iterating the {{domxref("RTCStatsReport")}} re
...
-  - : The total number of payload bytes sent (that is, the total number of bytes sent minus any headers, padding, or other administrative overhead) so far on this candidate pair.
+  - : An integer representing the total number of payload bytes sent on this candidate pair (the total number of bytes sent excluding any headers, padding, or other protocol overhead).
+- {{domxref("RTCIceCandidatePairStats/consentRequestsSent", "consentRequestsSent")}} {{optional_inline}}
+  - : An integer representing the total number of [STUN](/en-US/docs/Web/API/WebRTC_API/Protocols#stun) consent requests sent on this candidate pair.
 - {{domxref("RTCIceCandidatePairStats.currentRoundTripTime", "currentRoundTripTime")}} {{optional_inline}}
-  - : A floating-point value indicating the total time, in seconds, that elapsed between the most recently-sent STUN request and the response being received. This may be based upon requests that were involved in confirming permission to open the connection.
+  - : A real number representing the total time, in seconds, that elapsed between the most recently-sent STUN request and the response being received.
```
Comment (hamishwillee) [inline files/en-us/web/api/rtcicecandidatepairstats/index.md:33]:
```suggestion
  - : A number representing the total time, in seconds, that elapsed between the most recently-sent STUN request and the response being received.
```

**Comment by hamishwillee** [comment]
Thanks for the review. Ready for you to have another look.

**Comment by wbamberg** [comment]
> Tell me, would you assume that the value could include fractional seconds? 

Probably not. But TBH I don't know enough about the use case to know if this would be a problem.

The risk of using "real number" is that if I then see "number" somewhere else (like https://pr36527.content.dev.mdn.mozit.cloud/en-US/docs/Web/API/RTCIceCandidatePairStats/availableIncomingBitrate#value, which AFAICT could also be fractional) then I wonder what the difference is. So we could consistently say "integer" and "real number" (or "floating-point number"?) and that might be clearest (also with signed/unsigned variants). But I don't know if we want to go there.

I didn't see any guidelines about this (for instance here: https://developer.mozilla.org/en-US/docs/MDN/Writing_guidelines/Howto/Write_an_api_reference/Information_contained_in_a_WebIDL_file).

**Review by wbamberg — APPROVED** [review]
Thanks Hamish! Merging this but happy to continue [this conversation](https://github.com/mdn/content/pull/36527#issuecomment-2452361516) here or somewhere else.

**Comment by hamishwillee** [comment]
> ... happy to continue [this conversation](https://github.com/mdn/content/pull/36527#issuecomment-2452361516) here or somewhere else.

Thanks @wbamberg !

I've added a conversation here: https://github.com/orgs/mdn/discussions/754

It doesn't matter particularly about this use case: my concern is that the convention might lead to false assumptions in cases where it does matter. I might be overthinking it. Either way, the end result of the discussions should be an update to the template or guidelines, and that is no bad thing.
